Update documentation for some infra related processes (#294)
ngi-nix/ngipkgs: infra/README.md

## Rebuilding `makemake`

### Applying patches automatically

> By default, a [GitHub action](https://github.com/ngi-nix/ngipkgs/actions/workflows/makemake.yaml) deploys `makemake` automatically (on every push to `main`).

Reviewer: ```suggestion
By default, [a GitHub action deploys `makemake` automatically](https://github.com/ngi-nix/ngipkgs/actions/workflows/makemake.yaml) on every push to the `main` branch.
```

## Rebuilding `makemake`

### Applying patches automatically
By default, a [GitHub
action](https://github.com/ngi-nix/ngipkgs/actions/workflows/makemake.yaml)

> deploys `makemake` automatically on every merge. No manual actions are required

Reviewer 2: ```suggestion
deploys `makemake` automatically (on every push to main). No manual actions are required
```

Reviewer: @lorenzleutgeb we already discussed this, @Erethon will reformat to one sentence per line (and maybe add a note in the contributing guide).

Reviewer 2: I can't find the discussion, sorry.

Author: The discussion happened on a call last week, don't worry about it.

      ```
      nixos-rebuild switch --show-trace -L --flake /root/ngipkgs/#makemake
      ```

## Secret management
We use [sops-nix](https://github.com/Mic92/sops-nix) to manage secrets in NGIpkgs.
In order to access the secrets they must be encrypted with your public key and you must have the matching private key available in your system.

> [Sops](https://github.com/getsops/sops) must also be installed and configured.

Reviewer: ```suggestion
[SOPS](https://github.com/getsops/sops) must also be installed and configured.
```

### Viewing and editing secrets

> By default on Linux, sops expects the private keys to be in `$XDG_CONFIG_HOME/sops/age/keys.txt` or `$HOME/.config/sops/age/keys.txt`.

Reviewer: ```suggestion
By default on Linux, SOPS expects the private keys to be in `$XDG_CONFIG_HOME/sops/age/keys.txt` or `$HOME/.config/sops/age/keys.txt`.
```

> You can generate an age key using `age-keygen` or `ssh-to-age` to convert a SSH key to an age compatible one.

Reviewer: ```suggestion
You can generate an [age](https://github.com/FiloSottile/age) key using `age-keygen` or convert an SSH key with `ssh-to-age` to one compatible with age.
```

Once you've setup your key, you can run `sops -d <file-to-decrypt>` to view an encrypted file or `sops <file-to-edit>` to edit a file.

### Encrypting secrets for a new key

> To add a new person or a key edit `infra/.sops.yaml`, add the name of the person and their age public key under `.humans`.

Reviewer: ```suggestion
To add a new person or a key, edit `infra/.sops.yaml`, add the name of the person and their age public key under `.humans`.
```

      ```
      nixos-rebuild switch --show-trace -L --flake /root/ngipkgs/#makemake
      ```

## Secret management
We use [sops-nix](https://github.com/Mic92/sops-nix) to manage secrets in
NGIpkgs. In order to access the secrets they must be encrypted with your public
key and you must have the matching private key available in your system.
[Sops](https://github.com/getsops/sops) must also be installed and configured.

### Viewing and editing secrets
By default on Linux, sops expects the private keys to be in
`$XDG_CONFIG_HOME/sops/age/keys.txt` or `$HOME/.config/sops/age/keys.txt`. You
can generate an age key using `age-keygen` or `ssh-to-age` to convert a SSH key

> to an age compatible one. Once you've setup your key, you can run `sops -d

Reviewer 2: I dislike explicitly wrapped lines in Markdown. My editor will soft wrap for me anyways. Here it also breaks GitHub's syntax highlighting.

<file-to-decrypt>` to view an encrypted file or `sops <file-to-edit>` to edit a
file.

> ### Encrypting secrets for a new keυ

Reviewer: ```suggestion
### Encrypting secrets for a new key
```

> ### Encrypting secrets for a new keυ

Reviewer 2: Note that this process is vanilla SOPS. I am not sure it pays off to explicitly describe it versus just linking to SOPS docs.

Reviewer: Unless the sops docs have a summary of the same length we can link to directly, having two sentences on what to do is productive.

Author: [This](https://getsops.io/docs/#adding-and-removing-keys) is the upstream documentation on how to add more keys. Personally, I find there's value to document what sops usage in ngipkgs specifically looks like (hence the verbose `git diff` example).

To add a new person or a key edit `infra/.sops.yaml`, add the name of the
person and their age public key under `.humans`. Additionally, you need to add
their names to any files you want them to have access. This is done in
`infra/.sops.yaml` under the `creation_rules` section.

> This is an example diff of what this might look like:

Reviewer 2: Not sure I like how verbose that is. We could also refer to a commit via it's hash, or use `diff -U2` or so which would generate less lines.
